plugins/scheduled_task/lib/object_task_engine/KObjectTaskModifyEntryEngine.php

```diff
@@ -0,0 +1,158 @@
+<?php
+
+/**
+ * @package plugins.scheduledTask
+ * @subpackage lib.objectTaskEngine
+ */
+class KObjectTaskModifyEntryEngine extends KObjectTaskEntryEngineBase
+{
+	/**
+	 * @param KalturaBaseEntry $object
+	 */
+	function processObject($object)
+	{
+		/** @var KalturaModifyEntryObjectTask $objectTask */
+		$objectTask = $this->getObjectTask();
+		if (is_null($objectTask))
+			return;
+
+		KBatchBase::impersonate($object->partnerId);
+		$client = $this->getClient();
+		$metadataPlugin = KalturaMetadataClientPlugin::get($client);
+		$entryId = $object->id;
+		$outputMetadataProfileId = $objectTask->outputMetadataProfileId;
+		$outputMetadataArr = $objectTask->outputMetadata;
+		$inputMetadataProfileId = $objectTask->inputMetadataProfileId;
+		$inputMetadataArr = $objectTask->inputMetadata;
+		
+		$metadataFilter = new KalturaMetadataFilter();
+		$metadataFilter->metadataObjectTypeEqual = KalturaMetadataObjectType::ENTRY;
+		$metadataFilter->objectIdEqual = $entryId;
+		
+		if($outputMetadataProfileId != 0 && !empty($outputMetadataArr))
+		{
+			$entryResultForMetadataUpdate = $client->baseEntry->get($entryId);
+			
+			$metadataFilter->metadataProfileIdEqual = $outputMetadataProfileId;
+			self::updateMetadataObj($entryResultForMetadataUpdate, $metadataPlugin, $outputMetadataArr, $outputMetadataProfileId, $metadataFilter);
+		}
+		
+		KalturaLog::info("updating entry $entryId");
+		$entryObj = new KalturaBaseEntry();
+		
+		if($inputMetadataProfileId != 0 && !empty($inputMetadataArr))
+		{
+			$metadataFilter->metadataProfileIdEqual = $inputMetadataProfileId;
+			
+			$entryObj = self::updateEntryFromMetadata($metadataPlugin, $inputMetadataArr, $entryObj, $metadataFilter);
+		}
+
+		if(is_null($entryObj->userId))
+			$entryObj->userId = $objectTask->inputUserId;
+		if(is_null($entryObj->entitledUsersEdit))
+			$entryObj->entitledUsersEdit = $objectTask->inputEntitledUsersEdit;
+		if(is_null($entryObj->entitledUsersPublish))
+			$entryObj->entitledUsersPublish = $objectTask->inputEntitledUsersPublish;
+
+		try
+		{
+			$client->baseEntry->update($entryId, $entryObj);
+		}
+		catch(Exception $e)
+		{
+			KalturaLog::notice("problem with entry update entry id $entryId - " . $e->getMessage());
+		}
+		KBatchBase::unimpersonate();
+	}
+	
+	private static function updateMetadataObj(KalturaBaseEntry $entryResultForMetadataUpdate, &$metadataPlugin, $outputMetadataArr, $outputMetadataProfileId, $metadataFilter)
```

Comment:
Why static? (same goes for other methods)
@MosheMaorKaltura @tan-tan-kanarek agree that we usually don't use static methods in such cases?